sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodsShouldUseBaseTypes.cs

```diff
@@ -236,6 +237,17 @@ private INamedTypeSymbol FindOriginatingSymbol(ISymbol accessedMember)
                 : accessedMember.ContainingType;
         }
 
+        private bool IsEventHandlerImplementation(IMethodSymbol methodSymbol)

```

Comment:
there is an extension method SymbolHelper.IsProbablyEventHandler that does the same, shouldn't we change its implementation instead? Also perhaps we should check for the name of the second argument - args or e... Just to limit the possible false positives.

Reply:
It is a valid code not to have the second parameter inheriting from `EventArgs` so I would rather keep this method. I was tempted to change the other one but it might affect too many rules.